usefulUtilities/hiFiCalendar/clientScripts/meetingRoomSetup_app.js

```diff
@@ -0,0 +1,161 @@
+                // Gets the correct timezone offset for events current / in the future, which is why the -1 is needed.
+                timezoneOffset = (new Date().getTimezoneOffset() / MIN_PER_HR) * -1; 
+                roomConfig = data.connectionData;
+                Settings.setValue("roomConfigured", true);
+                Entities.callEntityServerMethod(TOKEN_SERVER_ID, "initializeRooms", [
+                    token, 
+                    refreshToken, 
+                    expireTime, 
+                    timezone,
+                    timezoneOffset, 
+                    JSON.stringify(roomConfig)
+                ]);
+                break;
+        }
+    }
+
+
+    // This function is the listener for messages on the message mixer related to this app.
+    var messageHandler = function(channel, message, senderUUID, localOnly) {
+        if (channel !== CHANNEL) {
+            return;
+        } else {
+            try {
+                message = JSON.parse(message);
+            } catch (e) {
+                console.log(e, "Could not parse message");
+                return;
+            }
+
+            if (message.TOKEN_SERVER_ID) {
+                TOKEN_SERVER_ID = message.TOKEN_SERVER_ID;
+            }
+
+            if (message.type === "TOKEN EXPIRED") {
+                Window.alert("The authorization token for your Google Calendar could not be refreshed.\n" + 
+                "Please open your calendar app and reauthorize to continue displaying calendar schedules.");
+                roomConfig = message.roomConfig;
+                roomConfigInfo = message.roomConfigInfo;
+            } else if (message.type === "STATUS UPDATE") {
+                if (message.tokenStatus) {
+                    Window.announcement("Access tokens are valid. No action required");
+                    roomConfig = message.roomConfig;
+                    roomConfigInfo = message.roomConfigInfo;
+                } else if (!message.tokenStatus && !message.roomConfigured) {
+                    roomConfigInfo = message.roomConfigInfo;
+                    Window.announcement("You have not set up any room schedules for this domain.");
+                }
+            } else if (message.type === "ERROR") {
+                // Commenting this out for right now because this was initially flooding the interface during testing
+                // Will bring back after battle testing for QA
+                
+                // Window.alert("There was an error, here's what we know:\n" + 
+                // message.entityName + "\nError: " + 
+                // message.errorMessage + "\nHappened during: " + 
+                // message.attemptedAction);
+                return;
+            } else if (message.type === "REFRESH SUCCESS") {
+                roomConfigInfo = message.roomConfigInfo;
+                Window.announcement("Token server successfully refreshed " + message.count + " times since preload.");
```

Comment:
Is this necessary?

Reply:
This one seemed like the only one that was helpful, at least so you know it is still working.

Reply:
yeah but does it have to be a `Window.announcement`? are our customers going to know what this means?

Reply:
I see what you are saying.  I suppose all of these could be console logs.  If there is an issue, than we can at least see a record of it if we need to in their logs.

Reply:
Right. If this is a `Window.announcement`, we have no recourse if people send us their logs with an issue; we'd be relying on the user to say "oh hey what's this weird announcement?" and then report it to us.